src/foam/nanos/fs/File.js

```diff
@@ -59,5 +69,48 @@ foam.CLASS({
         return url;
       }
     }
+  ],
+
+  methods: [
+    {
+      name: 'authorizeOnCreate',
+      javaCode: `
+        AuthService auth = (AuthService) x.get("auth");
```

Comment:
Not really this Pr's issue, but wouldn't it make more sense if all of these authorizeOnXYZ methods were just given the AuthService as the second arg, since they always have to look it up as the first thing anyway?